controllers/crds/cninode_controller.go

```diff
@@ -133,16 +135,12 @@ func (r *CNINodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// Add cluster name tag if it does not exist
 		val, ok := cniNode.Spec.Tags[config.VPCCNIClusterNameKey]
 		if !ok || val != r.clusterName {
-			if len(cniNodeCopy.Spec.Tags) != 0 {
-				cniNodeCopy.Spec.Tags[config.VPCCNIClusterNameKey] = r.clusterName
-			} else {
-				cniNodeCopy.Spec.Tags = map[string]string{
-					config.VPCCNIClusterNameKey: r.clusterName,
-				}
+			cniNodeCopy.Spec.Tags = map[string]string{
```

Comment:
Reassigning `Tags` makes me worry that we have to always maintain the order with tagging instance id later. Otherwise `Tags` are overwritten.